packages/oui-action-menu/src/action-menu-item.directive.js

```diff
@@ -0,0 +1,54 @@
+import controller from './action-menu-item.controller'
+
+const baseClass = 'oui-action-menu__item'
+const dividerClass = 'oui-action-menu__item_divider'
+const buttonClasses = 'oui-button oui-button_action-menu'
+
+const linkTemplate = (dividerClass, external) => `
+  <li class="${baseClass} ${dividerClass}">
+    <a
+      class="${buttonClasses}"
+      data-ng-href="{{::$ctrl.href}}">
+      <span data-ng-bind="::$ctrl.text"></span>
+      ${external ? '<i class="oui-icon oui-icon-external_link" aria-hidden="true"></i>' : ''}
+    </a>
+  </li>`
+
+const buttonTemplate = dividerClass => `
+  <li class="${baseClass} ${dividerClass}">
+    <button
+      class="${buttonClasses}"
+      type="button"
+      data-ng-click="$ctrl.onClick()"
+      data-ng-bind="::$ctrl.text"></button>
+  </li>`
+
+export default $compile => {
+  'ngInject'
+
+  return {
+    restrict: 'E',
```

Comment:
I don't use a component because my link is too complex for that.
I wonder if it was a good idea because I can use $postLink... any opinion other that "directives are evil"?
We decided that $postLink should be used to set root classes on component elements if necessary.

Reply:
Now it's a component.